test-runner/src/logging.rs

```diff
@@ -53,3 +58,91 @@ impl log::Log for StdOutBuffer {
 pub fn init_logger() -> Result<(), SetLoggerError> {
     log::set_logger(&*LOGGER).map(|()| log::set_max_level(LevelFilter::Info))
 }
+
+pub async fn get_mullvad_app_logs() -> LogOutput {
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    let settings_json = {
+        const SETTINGS_PATH: &str = "/etc/mullvad-vpn/";
+        let settings_json_path = Path::new(SETTINGS_PATH).join("settings.json");
+        read_to_string(&settings_json_path).await.map_err(|e| {
+            Error::Logs(format!(
+                "{}: {}",
+                settings_json_path.to_str().unwrap_or("unknown path"),
+                e
+            ))
+        })
+    };
+    #[cfg(target_os = "windows")]
+    let settings_json = {
+        mullvad_paths::get_default_settings_dir()
+            .and_then(|mut path| {
+                path.push("settings.json");
+                read_to_string(&path).map_err(|e| {
+                    Error::Logs(format!(
+                        "{}: {}",
+                        path.to_str().unwrap_or("unknown path"),
+                        e
+                    ))
+                })
+            })
+            .map_err(|e| Error::Logs(format!("{}", e)))
+    };
+
+    #[cfg(target_os = "linux")]
+    const LOGS_PATH: &str = "/var/log/mullvad-vpn/";
+    #[cfg(target_os = "macos")]
+    const LOGS_PATH: &str = "/var/log/mullvad-vpn/";
+    #[cfg(target_os = "windows")]
+    const LOGS_PATH: &str = r"C:\ProgramData\Mullvad VPN\";
+
+    let log_files = match list_logs(LOGS_PATH).await {
+        Err(e) => Err(e),
+        Ok(paths) => {
+            let mut log_files = Vec::new();
+            for path in paths {
+                let content = match read_to_string(&path).await {
+                    Ok(content) => content,
+                    Err(e) => {
+                        log_files.push(Err(Error::Logs(format!(
+                            "{}: {}",
+                            path.to_str().unwrap_or("unknown path"),
+                            e
+                        ))));
+                        continue;
+                    }
+                };
+                let log_file = LogFile {
+                    content,
+                    name: path,
+                };
+                log_files.push(Ok(log_file));
+            }
+            Ok(log_files)
+        }
+    };
+
+    LogOutput {
+        settings_json,
+        log_files,
+    }
+}
+
+async fn list_logs<T: AsRef<Path>>(log_dir: T) -> Result<Vec<PathBuf>, Error> {
+    let mut dir_entries = tokio::fs::read_dir(&log_dir).await.map_err(|e| {
+        Error::Logs(format!(
+            "{}: {}",
+            log_dir.as_ref().to_str().unwrap_or("unknown path"),
```

Comment:
Nit: You should be able to use `log_dir.as_ref().display()` here. Applies to other paths in the file as well.

Reply:
Fixed.